sklearn/utils/estimator_checks.py

```diff
@@ -1472,51 +1484,73 @@ def multioutput_estimator_convert_y_2d(name, y):
 
 
 @ignore_warnings(category=DeprecationWarning)
-def check_non_transformer_estimators_n_iter(name, estimator,
-                                            multi_output=False):
-    # Check if all iterative solvers, run for more than one iteration
-
-    iris = load_iris()
-    X, y_ = iris.data, iris.target
-
-    if multi_output:
-        y_ = np.reshape(y_, (-1, 1))
+def check_non_transformer_estimators_n_iter(name, Estimator):
+    # Test that estimators that are not transformers with an attribute
+    # max_iter, return the attribute of n_iter at least 1.
+
+    # These models are dependent on external solvers like
+    # libsvm and accessing the iter parameter is non-trivial.
+    not_run_check_n_iter = ['Ridge', 'SVR', 'NuSVR', 'NuSVC',
+                            'RidgeClassifier', 'SVC', 'RandomizedLasso',
+                            'LogisticRegressionCV', 'LinearSVC',
+                            'LogisticRegression']
+
+    # Tested in test_transformer_n_iter
+    not_run_check_n_iter += CROSS_DECOMPOSITION
+    if name in not_run_check_n_iter:
+        return
 
-    set_random_state(estimator, 0)
-    if name == 'AffinityPropagation':
-        estimator.fit(X)
+    # LassoLars stops early for the default alpha=1.0 the iris dataset.
+    if name == 'LassoLars':
```

Comment:
Good point. This check was performed prior to the PR and I just moved it from `tests/test_common.py` to `utils/estimator_checks.py`: https://github.com/scikit-learn/scikit-learn/pull/7677/commits/f534c3409b1a1e524e7c1e632c93b31d223d578e#diff-b6e2c29089c86706dc868c7bfd58c1d6L174

If this is not done, the respective test fails:

```
======================================================================
FAIL: /home/ajunge/Code/scikit-learn/sklearn/tests/test_common.py.test_non_meta_estimators:check_non_transformer_estimators_n_iter(LassoLars)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ajunge/local/anaconda3/envs/sklearn/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/ajunge/Code/scikit-learn/sklearn/utils/testing.py", line 830, in __call__
    return self.check(*args, **kwargs)
  File "/home/ajunge/Code/scikit-learn/sklearn/utils/testing.py", line 355, in wrapper
    return fn(*args, **kwargs)
  File "/home/ajunge/Code/scikit-learn/sklearn/utils/estimator_checks.py", line 1522, in check_non_transformer_estimators_n_iter
    assert_greater_equal(estimator.n_iter_, 1)
  File "/home/ajunge/Code/scikit-learn/sklearn/utils/testing.py", line 156, in assert_greater_equal
    assert a >= b, message
AssertionError: 0 is not greater than or equal to 1
>>  assert 0 >= 1, '0 is not greater than or equal to 1'
```

Do you suggest this could be handled differently?